avocado/core/result.py

```diff
@@ -133,12 +137,14 @@ def start_tests(self):
         Called once before any tests are executed.
         """
         self.tests_run += 1
+        self._result_events_dispatcher.map_method('pre_tests', self.job)
```

Comment:
First of all, thanks for reviewing and checking the remote runners, as they were neglected.

Then, I feel a bit sad because the calling of those hooks should live at the original places (as in V1/V2), but they can't because the remote runner don't implement all the steps/phases that the local runner do.

I was thinking about holding these changes, but working on the remote runners at this level may be a bit far off. So, we'd lose the conversion to the new style plugin loading/execution.

Let's go forward with this, and work on our last two plugin types: runners and loaders.  When working on runners, we should be able to define a basic interface that all runners should implement and move the result dispatching to the right location.